modules/core/shared/src/main/scala/gem/math/Angle.scala

```diff
@@ -260,6 +260,24 @@ trait AngleOptics extends OpticsHelpers { this: Angle.type =>
     )
   }
 
+  // Exact signed angles, scaled by moving the decimal point `scale` digits to the left
+  private def signedMicroarcsecondsScaled(scale: Int): SplitMono[Angle, BigDecimal] =
+    signedMicroarcseconds.imapB(_.underlying.movePointRight(scale).longValue, n => new java.math.BigDecimal(n).movePointLeft(scale))
```

Comment:
`imap` I suppose is invariant map.  What's `imapB`?

Reply:
Consider 

```scala
type Endo[A] = A => A
```

If we want an `Endo[B]` we would need to supply a `B => A` for the input on the left and `A => B` for the output on the right. `imap[B](f: A => B)(g: B => A): Endo[B]` does this. Types like `Endo` are called invariant functors because `A` appears in both covariant and contravariant positions.

Now consider

```scala
type Iso[A, B] = (A => B, B => A)
```

If we want an `Iso[A, C]` we need `B => C` on the right of the first function, and `C => B` on the left of the second function, as with `Endo`. But if we want `Iso[C, B]` we need the same kind of thing, with `C => A` and `A => C`, also as with `Endo`. So `Iso` is an invariant functor in two different ways.

`Iso` and other bidirectional optics have this 2-headed form so you have `imapA` for the left-side invariant functor, and `imapB` for the right-side one.

Does this make sense?

Reply:
Yes very much.  Thanks for the detailed explanation.